compiler/rustc_hir_analysis/src/hir_ty_lowering/object_safety.rs

```diff
@@ -214,7 +216,9 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
         // Erase the `dummy_self` (`trait_object_dummy_self`) used above.
         let existential_trait_refs = regular_traits.iter().map(|i| {
             i.trait_ref().map_bound(|trait_ref: ty::TraitRef<'tcx>| {
-                assert_eq!(trait_ref.self_ty(), dummy_self);
+                if !trait_ref.self_ty().references_error() {
+                    assert_eq!(trait_ref.self_ty(), dummy_self);
+                }
```

Comment:
I am a bit nervous about this change. It seems like the rest of the compiler ought to be able to assume that trait objects are represented "correctly".

Reply:
I reverted the change and instead only marked those generic params as errors that were actually erroneous